src/db.rs

```diff
@@ -292,29 +294,58 @@ impl DbInner {
     }
 
     async fn replay_wal(&self) -> Result<(), SlateDBError> {
+        async fn load_sst_iters(
+            db_inner: &DbInner,
+            sst_id: u64,
+        ) -> Result<(SstIterator<'_, Rc<SsTableHandle>>, u64), SlateDBError> {
+            let sst = Rc::new(
+                db_inner
+                    .table_store
+                    .open_sst(&SsTableId::Wal(sst_id))
+                    .await?,
+            );
+            let id = match &sst.id {
+                SsTableId::Wal(id) => *id,
+                SsTableId::Compacted(_) => return Err(SlateDBError::InvalidDBState),
+            };
+            let iter =
+                SstIterator::new_spawn(Rc::clone(&sst), db_inner.table_store.clone(), 1, 256, true)
+                    .await?;
+            Ok((iter, id))
+        }
+
         let wal_id_last_compacted = self.state.read().state().core.last_compacted_wal_sst_id;
-        let wal_sst_list = self
+        let mut wal_sst_list = self
             .table_store
             .list_wal_ssts((wal_id_last_compacted + 1)..)
             .await?
             .into_iter()
             .map(|wal_sst| wal_sst.id.unwrap_wal_id())
             .collect::<Vec<_>>();
         let mut last_sst_id = wal_id_last_compacted;
+        let sst_batch_size = 4;
+
+        let mut remaining_sst_list = Vec::new();
+        if wal_sst_list.len() > sst_batch_size {
+            remaining_sst_list = wal_sst_list.split_off(sst_batch_size);
+        }
+        let mut remaining_sst_iter = remaining_sst_list.iter();
+
+        // Load the first N ssts and instantiate their iterators
+        let mut sst_iterators = VecDeque::new();
         for sst_id in wal_sst_list {
```

Comment:
As I read the code, it looks like you're loading `remaining_sst_list` only if there are > `sst_batch_size` in the WAL. Otherwise, it remains empty.

Then, it looks like you instantiate SST iterators for all remaining WAL SSTs and put them into sst_iterators. Did you mean to use `remaining_sst_list` in the for loop here?

And also, I think if the `wal_sst_list.len()` is <= `sst_batch_size`, you should put all SSTs into `remaining_sst_list`? Otherwise it's empty.

Reply:
Ah, I misunderstood how [`split_off`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.split_off) works! Looks like it leaves the first N in `wal_sst_list`.